org.adempiere.base/src/org/compiere/acct/Doc_AllocationHdr.java

```diff
@@ -1869,16 +1875,33 @@ public boolean createEntries (MAcctSchema as, Fact fact, DocLine line)
 					{
 						//for sales actions
 						if (m_IsSOTrx) {
-							fact.createLine (line, m_DiscountAccount,
-								as.getC_Currency_ID(), amount, null);
-							fact.createLine (line, taxAcct,
-								as.getC_Currency_ID(), null, amount);
+							doctypecm = MDocType.getDocType(MDocType.DOCBASETYPE_ARCreditMemo);
```

Comment:
This is wrong here.
This line is getting the doctype for a credit memo, but there can be more than one and the method is returning the first.

Instead of comparing a doctype, it must check for the DocBaseType directly.

To make it fail I created a second "AR Credit Memo 2" document, simply copying it from "AR Credit Memo" - then the code here fails to realize it is a credit memo.

There is similar code below to fix.

Reply:
Hi @CarlosRuiz-globalqss 
yes, that can cause an error.
I will correct the part of code.
Many Thanks!
Best regards,
patric

Reply:
Changed to verify DocBaseType in DocAllocationHdr

Manual tested with different DockTypes
- DocType=AR Invoice 2 - DocBaseType=AR Invoice
- DocType=AR CreditMemo 2 - DocBaseType=AR CreditMemo

- DocType=AP Invoice 2 - DocBaseType=AP Invoice
- DocType=AP CreditMemo 2 - DocBaseType=AP CreditMemo